### ISSUE TITLE
Alt Modifier Stops Working After Latest macOS Sequoia Developer Beta

### ISSUE THREAD

**Issue by jcdiv47** [nikitabobko/AeroSpace#482]
Pretty much an identical issue as in #285 where the `alt` modifier appears to stop responding after MacOS Sequoia update. The issue was reported to be solved in the latest version of Aerospace at the time (June 26).

I must have started using Aerospace after that time because I am also in MacOS Sequoia and I have been enjoying Aerospace since my first experience up until yesterday. It appears that my `alt` modifier stops working in Aerospace after my MacBook Pro's auto update last night where my system updates to the latest MacOS Sequoia 15 Developer Beta(24B5035e).

I have made sure the following few things:

- updating Aerospace to version `0.14.2-Beta` and restarting the app but the problem still presists
- change `alt` to say `cmd` in Aerospace config and it works fine
- use Aerospace from command line, tested a couple of basic commands and it works fine
- my `alt` key is not broken, e.g., I can switch between MacOS desktops with `alt-left` and `alt-right`

Changing `alt` to `cmd` has too many conflicts in key bindings than I am willing to deal with and `ctrl` is just like way in the corner and I need to build up muscle memory all over again. Therefore, I really hope I can identify the problem and fix it here. Appreciate any help/advice in advance.

<details>
<summary> Here's my full Aerospace config...</summary>

```
# Place a copy of this config to ~/.aerospace.toml
# After that, you can edit ~/.aerospace.toml to your liking

# It's not necessary to copy all keys to your config.
# If the key is missing in your config, "default-config.toml" will serve as a fallback

# You can use it to add commands that run after login to macOS user session.
# 'start-at-login' needs to be 'true' for 'after-login-command' to work
# Available commands: https://nikitabobko.github.io/AeroSpace/commands
after-login-command = []

# You can use it to add commands that run after AeroSpace startup.
# 'after-startup-command' is run after 'after-login-command'
# Available commands : https://nikitabobko.github.io/AeroSpace/commands
after-startup-command = []

# Start AeroSpace at login
start-at-login = true

# Normalizations. See: https://nikitabobko.github.io/AeroSpace/guide#normalization
enable-normalization-flatten-containers = true
enable-normalization-opposite-orientation-for-nested-containers = true

# See: https://nikitabobko.github.io/AeroSpace/guide#layouts
# The 'accordion-padding' specifies the size of accordion padding
# You can set 0 to disable the padding feature
accordion-padding = 20

# Possible values: tiles|accordion
default-root-container-layout = 'tiles'

# Possible values: horizontal|vertical|auto
# 'auto' means: wide monitor (anything wider than high) gets horizontal orientation,
#               tall monitor (anything higher than wide) gets vertical orientation
default-root-container-orientation = 'auto'

# Possible values: (qwerty|dvorak)
# See https://nikitabobko.github.io/AeroSpace/guide#key-mapping
key-mapping.preset = 'qwerty'

# Mouse follows focus when focused monitor changes
# Drop it from your config, if you don't like this behavior
# See https://nikitabobko.github.io/AeroSpace/guide#on-focus-changed-callbacks
# See https://nikitabobko.github.io/AeroSpace/commands#move-mouse
# on-focused-monitor-changed = ['move-mouse monitor-lazy-center']
# on-focus-changed = "move-mouse window-lazy-center"

# Gaps between windows (inner-*) and between monitor edges (outer-*).
# Possible values:
# - Constant:     gaps.outer.top = 8
# - Per monitor:  gaps.outer.top = [{ monitor.main = 16 }, { monitor."some-pattern" = 32 }, 24]
#                 In this example, 24 is a default value when there is no match.
#                 Monitor pattern is the same as for 'workspace-to-monitor-force-assignment'.
#                 See: https://nikitabobko.github.io/AeroSpace/guide#assign-workspaces-to-monitors
[gaps]
inner.horizontal = 0
inner.vertical =   0
outer.left =       5
outer.bottom =     5
outer.top =        0
outer.right =      5

# 'main' binding mode declaration
# See: https://nikitabobko.github.io/AeroSpace/guide#binding-modes
# 'main' binding mode must be always presented
[mode.main.binding]

# All possible keys:
# - Letters.        a, b, c, ..., z
# - Numbers.        0, 1, 2, ..., 9
# - Keypad numbers. keypad0, keypad1, keypad2, ..., keypad9
# - F-keys.         f1, f2, ..., f20
# - Special keys.   minus, equal, period, comma, slash, backslash, quote, semicolon, backtick,
#                   leftSquareBracket, rightSquareBracket, space, enter, esc, backspace, tab
# - Keypad special. keypadClear, keypadDecimalMark, keypadDivide, keypadEnter, keypadEqual,
#                   keypadMinus, keypadMultiply, keypadPlus
# - Arrows.         left, down, up, right

# All possible modifiers: cmd, alt, ctrl, shift

# All possible commands: https://nikitabobko.github.io/AeroSpace/commands

# You can uncomment this line to open up terminal with alt + enter shortcut
# See: https://nikitabobko.github.io/AeroSpace/commands#exec-and-forget
# alt-enter = 'exec-and-forget open -n /System/Applications/Utilities/Terminal.app'

# See: https://nikitabobko.github.io/AeroSpace/commands#layout
alt-slash = 'layout tiles horizontal vertical'
alt-comma = 'layout accordion horizontal vertical'

# See: https://nikitabobko.github.io/AeroSpace/commands#focus
alt-h = 'focus left'
alt-j = 'focus down'
alt-k = 'focus up'
alt-l = 'focus right'

# See: https://nikitabobko.github.io/AeroSpace/commands#move
alt-shift-h = 'move left'
alt-shift-j = 'move down'
alt-shift-k = 'move up'
alt-shift-l = 'move right'

# See: https://nikitabobko.github.io/AeroSpace/commands#resize
alt-shift-minus = 'resize smart -50'
alt-shift-equal = 'resize smart +50'

# See: https://nikitabobko.github.io/AeroSpace/commands#workspace
alt-1 = 'workspace 1'
alt-2 = 'workspace 2'
alt-3 = 'workspace 3'
alt-4 = 'workspace 4'
alt-5 = 'workspace 5'
alt-b = 'workspace B'
alt-c = 'workspace C'
alt-m = 'workspace M'
alt-n = 'workspace N'
alt-t = 'workspace T'

# See: https://nikitabobko.github.io/AeroSpace/commands#move-node-to-workspace
alt-shift-1 = 'move-node-to-workspace 1'
alt-shift-2 = 'move-node-to-workspace 2'
alt-shift-3 = 'move-node-to-workspace 3'
alt-shift-4 = 'move-node-to-workspace 4'
alt-shift-5 = 'move-node-to-workspace 5'
alt-shift-b = 'move-node-to-workspace B'
alt-shift-c = 'move-node-to-workspace C'
alt-shift-m = 'move-node-to-workspace M'
alt-shift-n = 'move-node-to-workspace N'
alt-shift-t = 'move-node-to-workspace T'

alt-shift-enter = 'fullscreen'

# See: https://nikitabobko.github.io/AeroSpace/commands#workspace-back-and-forth
alt-tab = 'workspace-back-and-forth'
# See: https://nikitabobko.github.io/AeroSpace/commands#move-workspace-to-monitor
alt-shift-tab = 'move-workspace-to-monitor --wrap-around next'

# See: https://nikitabobko.github.io/AeroSpace/commands#mode
alt-shift-semicolon = 'mode service'

cmd-h = [] # Disable "hide application"
cmd-alt-h = [] # Disable "hide others"

# 'service' binding mode declaration.
# See: https://nikitabobko.github.io/AeroSpace/guide#binding-modes
[mode.service.binding]
esc = ['reload-config', 'mode main']
r = ['flatten-workspace-tree', 'mode main'] # reset layout
#s = ['layout sticky tiling', 'mode main'] # sticky is not yet supported https://github.com/nikitabobko/AeroSpace/issues/2
f = ['layout floating tiling', 'mode main'] # Toggle between floating and tiling layout
backspace = ['close-all-windows-but-current', 'mode main']

alt-shift-h = ['join-with left', 'mode main']
alt-shift-j = ['join-with down', 'mode main']
alt-shift-k = ['join-with up', 'mode main']
alt-shift-l = ['join-with right', 'mode main']

[[on-window-detected]]
if.app-id = 'net.kovidgoyal.kitty' # kitty
run = 'move-node-to-workspace T' # as in [T]erminal

[[on-window-detected]]
if.app-id = 'company.thebrowser.Browser' # Arc
run = 'move-node-to-workspace B' # as in [B]rowser

[[on-window-detected]]
if.app-id = 'com.google.Chrome' # Chrome
run = 'move-node-to-workspace B' # as in [B]rowser

[[on-window-detected]]
if.app-id = 'com.todesktop.230313mzl4w4u92' # Curosr
run = 'move-node-to-workspace C' # as in [C]ode

# [[on-window-detected]]
# if.app-id = 'com.apple.finder'
# run = 'move-node-to-workspace M' # as in [M]isc.

[[on-window-detected]]
if.app-id = 'md.obsidian'
run = 'move-node-to-workspace N' # as in [N]ote

[[on-window-detected]]
if.app-id = 'com.linnk.linnk' # Cubox
run = 'move-node-to-workspace N' # as in [N]ote
```
</details>
but I essentially use the default config so pretty much every key binding starts with `alt`.

## Checklist

- [x] I've searched for duplicates. My search queries were (please don't forget to search in closed issues too):
  - alt
  - option
  - Sequoia
- [x] I've read the documentation https://nikitabobko.github.io/AeroSpace/guide & https://nikitabobko.github.io/AeroSpace/commands
- [x] I've searched in the documentation. My search queries were:
  - update
  - alt
  - modifier

```
aerospace CLI client version: 0.14.2-Beta 0cb8dbdfc5ee73b8cbc200f175f467ebead55201
AeroSpace.app server version: 0.14.2-Beta 0cb8dbdfc5ee73b8cbc200f175f467ebead55201
```

**Comment by nikitabobko** [comment]
Can't reporduce in `15.1 Beta 3 (24B5035e)`. The problem is somewhere on your side

**Comment by mingsheng13** [comment]
@DieracDelta How did you manage to fix this? I'm new to aerospace and are trying the default config. My option key won't let me focus windows or move windows but it worked when I'm using the aerospace commands. I'm on Sonoma 14.6.1 btw.

**Comment by mingsheng13** [comment]
Omg, thats it! I was running yabai with skhd before and I forgot to turn it off. Thanks man.

**Comment by moshialam** [comment]
Hello everyone, 
In my case the option stopped working a couple of days back. I had uninstalled skhd and yabai. But even after checking and reloading the config, the problem persists. I had not installed anything new recently, and have been on MacOS Sequoia since its launch.  
Is there anyway to identify where the problem is? Thanks!

**Comment by moshialam** [comment]
RIght. Hence I get `zsh: command not found: skhd`

**Comment by mingsheng13** [comment]
Have you tried to restart your computer

**Comment by moshialam** [comment]
I did not--it just started working again today on its own!  All I had done in between was quite aerospace and restart again a bunch of times. But I was doing that before too and it had not worked. 

I truly don't know what happened here, but glad its working back again! :-)
Thanks everyone!

**Comment by SimonOneNineEight** [comment]
i met the same issue one day after I updated my macbook to macOS sequoia. I tried to restart my laptop multiple times, reinstall aerospace, or restart aerospace, but it all failed to fix the problem.

I really believe it is the `alt` key problem because it works well when I temporary change my keymap from alt to cmd.

Does anyone encounters the same issue?

(btw I don't have skhd installed)

**Comment by mingsheng13** [comment]
> i met the same issue one day after I updated my macbook to macOS sequoia. I tried to restart my laptop multiple times, reinstall aerospace, or restart aerospace, but it all failed to fix the problem.
> 
> I really believe it is the `alt` key problem because it works well when I temporary change my keymap from alt to cmd.
> 
> Does anyone encounters the same issue?
> 
> (btw I don't have skhd installed)

This might be helpful.
https://apple.stackexchange.com/questions/104074/find-which-process-receives-a-given-keyboard-command

**Comment by jasmine-nguyen** [comment]
Can confirm I have the same issue after upgrading to Sequoia 15.1.1 (24B91). I never used yabai or skhd before so they should not be the culprit. I'm gonna try to downgrade to Sonoma and see if it fixes the issue.

**Comment by sschuldenzucker** [comment]
Hey guys, I'm also having this problem since I updated to Seqoia, but *only with kitty*. If kitty is focused, my `alt-shift-...` keyboard shortcuts do nothing. My `alt-shift-ctrl-...` shortcuts work. If kitty is not focused, everything works as normal. This does not happen with any other app (I've tested the default Terminal, Arc browser, slack)

I've never used skhd or yabai.

Aerospace version: 0.15.2-Beta b6cf82
Kitty version: 0.37.0

**Comment by sschuldenzucker** [comment]
@kaynetik & for the record: your instructions didn't help me but turns out that kitty's "secure text input" was on for some reason. Deactivating that (`cmd-option-s` in kitty by default, seems persistent) fixed the issue. Not sure if the Sequoia update activated that or whatever. See https://github.com/kovidgoyal/kitty/issues/8108#issuecomment-2525670208

**Comment by filip-krasinski** [comment]
Same issue, fresh macOS install... not a single shortcut works.

```
aerospace CLI client version: 0.16.2-Beta 21336ad382539b35fdc94b4fbd55408e10b101f8
AeroSpace.app server version: Unknown. The server is not running
macOS Sequoia Version 15.1 (24B2083)
```

**Comment by bashfulrobot** [comment]
I also have this issue, but I believe it is due to Apple disabling items like `alt` due to "malware". 

https://developer.apple.com/forums/thread/763878?answerId=804374022#804374022

How have others been working around this?

Cheers.

**Comment by djeikyb** [comment]
the "set it to start at login, then reboot" procedure fixed my <kbd>opt</kbd>+<kbd>`</kbd> (option + backtick) shortcut in moom, too. worked fine in sonoma, sequoia broke it

**Comment by inigourrestarazu** [comment]
> Same issue after upgrading to Sequoia 15.4.1 (24E263).
> 
> The problem is exclusively with `alt` key.
> 
> This solution did not work for me, though [#482 (comment)](https://github.com/nikitabobko/AeroSpace/issues/482#issuecomment-2524967747)

Okay, I changed all the modifiers to cmd... Saved, and reloaded config file, changed them back to alt, save and reload config, and voilá. It makes no sense to me, but it worked

**Comment by aslak01** [comment]
> [@kaynetik](https://github.com/kaynetik) & for the record: your instructions didn't help me but turns out that kitty's "secure text input" was on for some reason. Deactivating that (`cmd-option-s` in kitty by default, seems persistent) fixed the issue. Not sure if the Sequoia update activated that or whatever. See [kovidgoyal/kitty#8108 (comment)](https://github.com/kovidgoyal/kitty/issues/8108#issuecomment-2525670208)

Thank you! I thought I was going crazy!

**Comment by FrontierPsychiatrist** [comment]
I am having the same problem. I installed AeroSpace on Friday and it worked. I restarted my Mac today (Tuesday), after restart the alt key stopped working in AeroSpace - only special characters are inserted. I don't assume it's a problem with AeroSpace, but I have a lot of trouble finding a solution that makes the key work again.

**Comment by FrontierPsychiatrist** [comment]
@inigourrestarazu I am not using `kitty`. Secure text input in Terminal.app is turned off.

**Comment by inigourrestarazu** [comment]
Sorry, I think I pointed you to the wrong source, and I cannot find the right one either. I would check that Input monitoring is allowed for Aerospace (`Settings > Privacy & Security > Input Monitoring`). In my case, at some point the input monitoring for Aerospace was disabled. Hope that helps :)

**Comment by FrontierPsychiatrist** [comment]
That did fix the issue! It wasn't just disabled, though - I had to manually re-add AeroSpace to Input Monitoring. Which is odd given that I even re-installed it.

**Comment by inigourrestarazu** [comment]
@FrontierPsychiatrist you're right! That's what I did ---I did not remember properly. And, yeah, I've been using Aerospace for a longish time, and it has happened to me too

**Comment by nvimts** [comment]
> > Same issue after upgrading to Sequoia 15.4.1 (24E263).
> > The problem is exclusively with `alt` key.
> > This solution did not work for me, though [#482 (comment)](https://github.com/nikitabobko/AeroSpace/issues/482#issuecomment-2524967747)
> 
> Okay, I changed all the modifiers to cmd... Saved, and reloaded config file, changed them back to alt, save and reload config, and voilá. It makes no sense to me, but it worked

Saved my day!

**Comment by FrontierPsychiatrist** [comment]
And now 15.5 broke it again - and this time it's still listed in "Input Monitoring" :( The "change-config-and-reload" trick is also not doing it.

**Comment by nvimts** [comment]
> And now 15.5 broke it again - and this time AeroSpace is still listed in "Input Monitoring" :( The "change-config-and-reload" trick is also not doing it.

I just changed commonly used commands to those without using option/alt key. Not ideal but functional.

**Comment by pstephenwille** [comment]
same, need my Alt key to work on Chromium.  I switched to Unicode char set under Keyboard input and Intellij will honor the Alt key but not Chromeium

**Comment by AMulabeg** [comment]
I'm dealing with the opposite issue actually. I want to use CMD but for some reason moving between windows doesn't work. Everything else works without issues (Moving to different spaces, moving windows to different spaces etc). I have double checked to see if there are any conflicting keys, but that is not the case

**Comment by ravihlb** [comment]
I found this thread when trying to fix a similar issue - option key started producing special characters on input fields and shortcuts didn't work, pretty much out of nowhere. I fixed this by restarting through the Apple icon on the top left -> Restart... and **UNCHECKING "Reopen windows when logging back in"**. Had to uncheck it otherwise the issue persisted after rebooting.

**Comment by Aphoh** [comment]
I think I found the root of this issue! It's a feature called secure keyboard input. I followed the instructions [here](https://alexwlchan.net/2021/secure-input/), ran the script, killed the process using this input method, and my alt shortcuts came back! I've found several different pieces of software that spin this up. @nikitabobko (sorry for the ping, I love your software) is there any way `Aerospace` can get around this feature?

**Comment by nikitabobko** [comment]
> is there any way Aerospace can get around this feature?

It's macOS security feature, and no app can bypass it.

But apps can detect when secure input is on. The next AeroSpace version will show a small popup in the screen's bottom right corner to inform users that it's prevented from handling global keys: https://github.com/nikitabobko/AeroSpace/issues/1486

**Comment by vitalNohj** [comment]
Hey just wanted to add that I got the popup for the secure keyboard input recently.

When I'm working with SSH and Cursor -> The SSH password prompt uses this and prevents Alt usage.

**Comment by dhgwilliam** [comment]
this is the only thing that reliably works for me on Sequoia after upgrade to 15.7.3, and in particular, it seems I have to ACTIVELY UNCHECK "Reopen windows..."

 in other words, if it is unchecked already, I have to check it and then uncheck it.

> I found this thread when trying to fix a similar issue - option key started producing special characters on input fields and shortcuts didn't work, pretty much out of nowhere. I fixed this by restarting through the Apple icon on the top left -> Restart... and **UNCHECKING "Reopen windows when logging back in"**. Had to uncheck it otherwise the issue persisted after rebooting.

**Comment by nelaaro** [comment]
My 2 cents, 
This happened after a config reload. 

macos alt opt key not working as expected making funny symbols ¡™£¢§ˆ
so alt+1 = ¡ alt+2=™ .. etc

I tried to switch input methods to fix this no luck. 

I went to add aerospace to input monitoring privileges. Which required I quit and restart aerospace. 
This then fixed the wried input mode breaking the alt/opt key.
Cause was linked to some strange state aerospace entered during config reload and alt-bind and service mode. 
Some edge case that mac ends up putting into some kind of input 

So fix for me was to quit aerospace and start it up again.
